cryptol-remote-api/python/cryptol/commands.py

```diff
@@ -161,14 +183,62 @@ def __init__(self, connection : HasProtocolState, expr : Any) -> None:
     def process_result(self, res : Any) -> Any:
         return res['type schema']
 
+
 class SmtQueryType(str, Enum):
     PROVE = 'prove'
     SAFE  = 'safe'
     SAT   = 'sat'
 
-class CryptolProveSat(argo.Command):
+class SmtResult(ABC):
+    """Abstract class for SMT query results."""
+    qtype: SmtQueryType
+    
+    def __init__(self, qtype : SmtQueryType) -> None:
+        if qtype in [SmtQueryType.PROVE, SmtQueryType.SAFE, SmtQueryType.SAT]:
+            self.qtype = qtype
+        else:
+            raise ValueError("Unknown SMT query type: " + qtype)            
+
+class SmtUnsatResult(SmtResult):
+    def __init__(self, qtype : SmtQueryType) -> None:
+        super(SmtUnsatResult, self).__init__(qtype)
+
+    def __bool__(self) -> bool:
+        """An unsat result is a success for PROVE and SAFE queries but a
+        failure for SAT queries. Thus, the object is truthy iff qtype != SAT.
+        """
+        return self.qtype != SmtQueryType.SAT
+
+class SmtSatResult(List[CryptolPython], SmtResult):
+    def __init__(self, qtype : SmtQueryType, cxs : List[CryptolPython]) -> None:
```

Comment:
I think these parameter names would be clearer if they were a hair more verbose. I can deduce `qtype` means `query_type`, but `cxs` is making me scratch my head a little, and I can't yet decide if I'm having a bonehead moment or it really is unclear xD

Reply:
Nope, it's just plain old bad naming - that was supposed to be short for "counterexample" D:

This class no longer exists, however, so this no longer needs to be fixed.